code/_helpers/sound_effects.dm

```diff
@@ -0,0 +1,27 @@
+/*
+Current icons:
+default - full size circle
+small - small circle
+boom - Jagged circle
+
+Source variable is used for mobs that make sounds to remove them from all those who these sounds should be shown to.
+So you don't see your own footsteps. Could use oview, but meh
+~Tsuru
+*/
+
+/proc/show_sound_effect(turf/T, mob/source, soundicon = "default")
```

Comment:
what's up with mob/source
you sometimes set it to null and sometimes you dont include it whatsoever
also shouldn't the source be /atom to accomodate, for instance, airlocks making noise?

Reply:
Source is for mob making sound to exclude THE MOB's client out of client list. You don't need to exclude airlock's client from mobs to flick overlay.